addons/pos_self_order/controllers/orders.py

```diff
@@ -95,28 +95,27 @@ def update_existing_order(self, order, access_token, table_identifier):
 
         if not order_sudo:
             raise Unauthorized("Order not found in the server !")
-        elif order_sudo.state != 'draft':
-            raise Unauthorized("Order is not in draft state")
```

Comment:
Why would you need an error for that ? The order is saved on the server if possible, and you update the client cache with the answer from the server, so the customer has the same order data than what is saved on the server. If you want to inform the customer that his changes couldn't be applied because the order wasn't editable, the safest approach is to compare what is returned by the server and what you have in the customer cache. What matters is not if the server changed the data of the order on the database, what matters is if the data that is finally saved on the server database is the same or is different than what is currently saved in the customer cache.

Reply:
Previously, when the client received an Unauthorized or NotFound error, the orders linked to this error were deleted from the front-end. 

If one of these errors occurs, the data is considered obsolete. Before this could happen, in the case of bad access_token -> delete, in the case of trying to edit an order in a state other than draft -> delete. 

In this kind of error, we don't really care about aligning front-end and back-end data, because the user isn't supposed to always have access to them.

So that's why the error is appropriate.

Reply:
Why would you "delete" (remove from the self-order frontend) an order that is correctly paid and saved on the server ? It should be visible but uneditable. For online payment, I need the data of the saved order, and your error what forcing me to do another request to the server for no reason (or to don't use your method at all, creating my owns, and having to handle the self-order frontend cache update, duplicating code...).
Here it's not a matter of access, only a matter of is the order editable on the server.

Reply:
You should never use an order that receive an error like that for online payment.

Reply:
I don't, see: https://github.com/odoo/odoo/pull/127869/files#diff-cc6b90485046b5f0a630784dbd290c5ddfdb6a41b422314bb5d85adff5444d3aR58

Reply:
Removed the changes in the self-order web controller.